polling_stations/apps/data_importers/data_quality_report.py

```diff
@@ -309,224 +284,210 @@ class DataQualityReportBuilder:
+                end_section=True,
             )
 
-            self.report.append(
-                ANSI.ok_bold(
-                    f" - As % of uprns in addressbase  : {round(100 * station_ids / uprns_in_council_area, 1)}%"
-                )
+            table.add_row(
+                " - As % of uprns in addressbase",
+                f"{round(100 * station_ids / uprns_in_council_area, 1)}%",
             )
             if self.csv_rows:
-                self.report.append(
-                    ANSI.ok_bold(
-                        f" - As % of rows in council csv   : {round(100 * station_ids / self.csv_rows, 1)}%"
-                    )
-                )
-            self.report.append("----------------------------------")
-            self.report.append(
-                ANSI.ok(
-                    "   - valid station id refs       : %i"
-                    % (address_report.get_addresses_with_valid_station_id_ref()),
+                table.add_row(
+                    " - As % of rows in council csv",
+                    f"{round(100 * station_ids / self.csv_rows, 1)}%",
                 )
+            table.add_row(
+                " - valid station id refs",
+                str(address_report.get_addresses_with_valid_station_id_ref()),
+                style="green",
             )
-            self.report.append(
-                ANSI.warning(
-                    "   - invalid station id refs     : %i"
-                    % (address_report.get_addresses_with_invalid_station_id_ref()),
-                )
+            table.add_row(
+                " - invalid station id refs",
+                str(address_report.get_addresses_with_invalid_station_id_ref()),
+                style="yellow",
             )
         else:
-
-            self.report.append(ANSI.warning("NO POLLING STATIONS ASSIGNED...      "))
-        self.report.append("----------------------------------")
-        self.report.append("\n")
+            table.add_row(
+                "NO POLLING STATIONS ASSIGNED", "", style="yellow", end_section=True
+            )
+        return table
 
     def build_report(self):
-        self.build_header()
-        self.build_district_report()
-        self.build_station_report()
-        self.build_address_report()
+        self.report.add_row(Panel(Text("DATA QUALITY REPORT", justify="center")))
+        if self.expecting_districts:
+            self.report.add_row(self.build_district_report())
+        self.report.add_row(self.build_station_report())
+        self.report.add_row(self.build_address_report())
 
     def output_console_report(self):
-        print("\n".join(self.report))
+        console = Console()
+        console.print(self.report)
 
     def generate_string_report(self):
-        return ANSI.remove_escapes("\n".join(self.report))
+        recorder = Console(record=True)
+        with recorder.capture():
+            recorder.print(self.report)
+        return recorder.export_text()
```

Comment:
Here I am using a special console object to capture the output instead of sending it to the terminal. Even if you switch to a global console object, you would still want to use a local object here.